src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs

```diff
@@ -8602,6 +8810,1964 @@
+                // struct S1(Color Color)
+                Diagnostic(ErrorCode.WRN_UnreadRecordParameter, "Color").WithArguments("Color").WithLocation(2, 17),
+                // (4,20): error CS9500: Cannot use primary constructor parameter 'Color Color' in this context.
+                //     static int F = Color.M1(new S1());
+                Diagnostic(ErrorCode.ERR_InvalidPrimaryConstructorParameterReference, "Color").WithArguments("Color Color").WithLocation(4, 20)
+                );
+
+            Assert.Empty(comp.GetTypeByMetadataName("S1").InstanceConstructors.OfType<SynthesizedPrimaryConstructor>().Single().GetCapturedParameters());
+        }
+
+        [Fact]
+        public void ParameterCapturing_044_ColorColor_MemberAccess_InstanceAndStatic_InStaticMethod_Method()
+        {
+            var source = @"
+struct S1(Color Color)
+{
+    public static string Test() => Color.M1(new S1());
+}
+
+class Color
+{
+    public string M1(S1 x, int y = 0) => ""instance"";
+    
+    public static string M1<T>(T x) where T : unmanaged => ""static"";
+}
+
+class Program
+{
+    static void Main()
+    {
+        System.Console.WriteLine(S1.Test());
+    }
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseExe);
+
+            CompileAndVerify(comp, expectedOutput: @"static").VerifyDiagnostics(
+                // (2,17): warning CS8907: Parameter 'Color' is unread. Did you forget to use it to initialize the property with that name?
+                // struct S1(Color Color)
+                Diagnostic(ErrorCode.WRN_UnreadRecordParameter, "Color").WithArguments("Color").WithLocation(2, 17)
+                );
+
+            Assert.Empty(comp.GetTypeByMetadataName("S1").InstanceConstructors.OfType<SynthesizedPrimaryConstructor>().Single().GetCapturedParameters());
+        }
+
+        [Fact]
+        public void ParameterCapturing_045_ColorColor_MemberAccess_InstanceAndStatic_InStaticMethod_Method()
+        {
+            var source = @"
+struct S1(Color Color)
+{
+    public static int Test() => Color.M1(new S1());
+}
+
+class Color
+{
+    public int M1(S1 x) => 0;
+    
+    public static int M1<T>(T x) where T : unmanaged => 0;
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseDll);
+
+            comp.VerifyEmitDiagnostics(
+                // (2,17): warning CS8907: Parameter 'Color' is unread. Did you forget to use it to initialize the property with that name?
+                // struct S1(Color Color)
+                Diagnostic(ErrorCode.WRN_UnreadRecordParameter, "Color").WithArguments("Color").WithLocation(2, 17),
+                // (4,33): error CS9500: Cannot use primary constructor parameter 'Color Color' in this context.
+                //     public static int Test() => Color.M1(new S1());
+                Diagnostic(ErrorCode.ERR_InvalidPrimaryConstructorParameterReference, "Color").WithArguments("Color Color").WithLocation(4, 33)
```

Comment:
>Diagnostic(ErrorCode.ERR_InvalidPrimaryConstructorParameterReference, "Color").WithArguments("Color Color").WithLocation(4, 33) [](http://example.com/codeflow?start=16&length=127)

This error feels unexpected. The parameter is not used here. I would expect this code to compile and call the static method.

Reply:
> This error feels unexpected. The parameter is not used here. I would expect this code to compile and call the static method.

Language rules do not agree with these expectations
[https://sharplab.io/#v2:C4LglgNgPgzsBOBXAxsABAZQIwFgBQA3vmiWgAIDM5WAbGmAHboAqApnABQDCA9hD/DS9+8AJRoAvAD4hfAQDoAslg4NWAd0wrRogNz4AvvnxkATLJH4ieUuSqN0yjtjQAPcdLQAGfTdLFSSmo6BzRlAB5mKQ5mN3F1AAtWeFY0WJA0RAYAWwBDBlyAc1YAE0kZH0N8IA===](https://sharplab.io/#v2:C4LglgNgPgzsBOBXAxsABAZQIwFgBQA3vmiWgAIDM5WAbGmAHboAqApnABQDCA9hD/DS9+8AJRoAvAD4hfAQDoAslg4NWAd0wrRogNz4AvvnxkATLJH4ieUuSqN0yjtjQAPcdLQAGfTdLFSSmo6BzRlAB5mKQ5mN3F1AAtWeFY0WJA0RAYAWwBDBlyAc1YAE0kZH0N8IA===)
```
struct S1
{
    public static int Test(Color Color) => Color.M1(new S1());
}

class Color
{
    public int M1(S1 x) => 0;
    
    public static int M1<T>(T x) where T : unmanaged => 0;
}
```

Observed:
```
    .method public hidebysig static 
        int32 Test (
            class Color Color
        ) cil managed 
    {
        .custom instance void System.Runtime.CompilerServices.NullableContextAttribute::.ctor(uint8) = (
            01 00 01 00 00
        )
        // Method begins at RVA 0x20a0
        // Code size 16 (0x10)
        .maxstack 2
        .locals init (
            [0] valuetype S1
        )

        IL_0000: ldarg.0
        IL_0001: ldloca.s 0
        IL_0003: initobj S1
        IL_0009: ldloc.0
        IL_000a: callvirt instance int32 Color::M1(valuetype S1)
        IL_000f: ret
    } // end of method S1::Test
```